src/main/java/uk/gov/digital/ho/hocs/domain/queue/migration/MigrationService.java

```diff
@@ -38,40 +44,121 @@ public class MigrationService {
+                                attachment.getType(),
+                                attachment.getDocumentPath(),
+                                caseId);
+                documentClient.createDocument(caseId, document);
+            }
+            messageLogService.updateCaseUuidAndStatus(requestData.getCorrelationId(), caseId, Status.CASE_DOCUMENT_CREATED);
+            log.info("Created case attachments for migrated case {}", caseId);
+        } catch (Exception e) {
+            messageLogService.updateStatus(requestData.getCorrelationId(), Status.CASE_DOCUMENT_FAILED);
+            throw new ApplicationExceptions.DocumentCreationException(e.getMessage(), LogEvent.CASE_DOCUMENT_CREATION_FAILURE);
+        }
+        return ResponseEntity.ok().build();
     }
 
     CreateMigrationCaseRequest composeMigrateCaseRequest(MigrationData migrationData, MigrationCaseTypeData migrationCaseTypeData) {
         Map<String, String> initialData = Map.of(CHANNEL_LABEL, migrationCaseTypeData.getOrigin());
 
-        MigrationComplaintCorrespondent primaryCorrespondent = getPrimaryCorrespondent(migrationData.getPrimaryCorrespondent());
-        List<MigrationComplaintCorrespondent> additionalCorrespondents = getAdditionalCorrespondents(migrationData.getAdditionalCorrespondents());
-        List<CaseAttachment> caseAttachments = getCaseAttachments(migrationData.getCaseAttachments());
-
         return new CreateMigrationCaseRequest(
                 migrationData.getComplaintType(),
                 migrationData.getDateReceived(),
                 initialData,
-                "MIGRATION",
+                "MIGRATION");
+    }
+
+    CreateMigrationCorrespondentRequest composeMigrationCorrespondentRequest(UUID caseId, UUID stageId, MigrationData migrationData) {
+        MigrationComplaintCorrespondent primaryCorrespondent = getPrimaryCorrespondent(migrationData.getPrimaryCorrespondent());
+        List<MigrationComplaintCorrespondent> additionalCorrespondents = getAdditionalCorrespondents(migrationData.getAdditionalCorrespondents());
+
+        return new CreateMigrationCorrespondentRequest(
+                caseId,
+                stageId,
                 primaryCorrespondent,
-                additionalCorrespondents,
-                caseAttachments);
+                additionalCorrespondents
+        );
+    }
+
+    List<CaseAttachment> getCaseAttachments(UUID caseId, MigrationData migrationData) {
+        return getCaseAttachments(migrationData.getCaseAttachments());
+    }
+
+    public List<CaseAttachment> getCaseAttachments(String attachmentsJson) {
+        try {
+            List<CaseAttachment> caseAttachments = objectMapper.convertValue(
+                    objectMapper.readValue(attachmentsJson, JSONArray.class),
+                    new TypeReference<>() {
+                    });
+            return caseAttachments;
+        } catch (Exception e) {
+            return Collections.emptyList();
```

Comment:
This should probably at least log an error but should it also store the failure in the migration status database?